pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/SegmentCompletionManager.java

```diff
@@ -608,8 +618,17 @@ public boolean isDone() {
     private SegmentCompletionProtocol.Response commit(String instanceId, long offset) {
       long allowedBuildTimeSec = (_maxTimeAllowedToCommitMs - _startTimeMs)/1000;
       LOGGER.info("{}:COMMIT for instance={} offset={} buldTimeSec={}", _state, instanceId, offset, allowedBuildTimeSec);
-      return new SegmentCompletionProtocol.Response(new SegmentCompletionProtocol.Response.Params().withOffset(offset)
-          .withBuildTimeSeconds(allowedBuildTimeSec).withStatus( SegmentCompletionProtocol.ControllerResponseStatus.COMMIT));
+      if (_isSplitCommitEnabled) {
+        return new SegmentCompletionProtocol.Response(
```

Comment:
you shoiuld pull out the common parts outside. suggest:

SegmetCompletinProtocol.Response.Params params = new Params().with(offset).withBuildTimeSeconds().withStatus().withSPlitCommit();
if (_isSplitCommitEnabled) {
params.withControlerVipUrl();
}
return new SegmentCompletionProtocol.Response(params)